player.class.js

```diff
@@ -0,0 +1,92 @@
+import { c } from "./view.js";
+
+class Player {
+    constructor(x, y){
+        this.sx = 0;
+        this.sy = 0;
+        this.sw = 756.3;
+        this.sh = 340.8;
+        this.x = x;
+        this.y = y;
+        this.w = 200;
+        this.h = 100;
+        this.image = new Image();
+        this.image.src = "images/submarine.png";
+        this.interval;
+    }
+    render()  {
+        if (this.sx == 0 && this.sy == 0){
+            c.drawImage(this.image, this.sx, this.sy, this.sw, this.sh, this.x, this.y, this.w,this.h);
+            this.sx = 770;
+        }
+        else if (this.sx == 0 && this.sy == 344.4) {
+            c.drawImage(this.image, this.sx, this.sy, this.sw, this.sh, this.x, this.y, this.w,this.h); 
+        }
+        else {
+            c.drawImage(this.image, this.sx, this.sy, this.sw, this.sh, this.x, this.y, this.w,this.h);
+            this.sx = 0; 
+        }
+    } 
+
+    // Player animation
+    animation(direction) {
+        
+        clearInterval(this.interval); //clear unfinished interval
+        this.interval = setInterval(()=> { 
```

Comment:
tu pri tomto nemoze nastat nejaky stav, kedy ho pomocou tych if neodchytis ?? alebo sa to ma takto spravat stale ?? ...ze co by sa stalo ak by bol direction == "upleft" ale this.y by bolo < 70 ??